### ISSUE TITLE
sf-client network list has no --namespace option

### ISSUE THREAD

**Issue by mandoonandy** [shakenfist/client-python#73]
Should allow listing by namespace.

```
$ sf-client network list --help
Usage: sf-client network list [OPTIONS]

  List networks

Options:
  -a, --all  Include networks in error and deleted networks
  --help     Show this message and exit.
```

**Comment by mikalstill** [comment]
In fact, most calls lack a namespace argument. I think this is more complicated than I want it to be though -- at the moment we only support specifying a namespace if you're in the system namespace. Is that really worth adding something to every call? Or do we need a more complicated namespace permissions model?

**Comment by mandoonandy** [comment]
This is a good question. 

Should namespaces just be a separation tool or should they be a proper tenant separation?
Do we want a full RBAC model or should each SF cluster be used by cooperating trusted users.

At the moment (today), I like the idea of separate namespaces for separating jobs because I can use a single `clean` command to delete all the instances and networks in that namespace.

So maybe a tenant permissions model goes over the top of namespaces? (k8s style)

**Comment by mikalstill** [comment]
Yeah, I like the simplicity of what we have now. What I really want to do is something like:

- you create a system client (if you're in the system namespace)
- system clients can create sub-clients which are in another namespace
- remove all the namespace arguments on individual calls

Something like:

```
system_client = apiclient.Client()
foo_client = system_client.subclient('foo')
```

When the subclient makes a call, how does the API server know that its authorized? Maybe _actual_request_url() could insert a header or something?

**Comment by mikalstill** [comment]
What about an `X-ShakenFist-Namespace` header which only works if you're authed in the system namespace?

**Comment by mandoonandy** [comment]
I'm not sure about the actual efficiency of sending the namespace in the header vs as a parameter. Each call still needs to check the namespace for the particular parameter. I think you are also suggesting further change?

In the case where we have multiple CI jobs running plus a couple of people working in the same cluster - a perfect example of the SF use case - it would be great to just list the instances/networks in a particular namespace. As a user, I want to quickly create a namespace, test an idea for a few days then clean away everything in the namespace. I don't want to inadvertently interfere with other work or CI jobs but still have a quick clean up option.
